src/kinsol/kinsol.c

```diff
@@ -3010,50 +2975,148 @@ static int KINFP(KINMem kin_mem)
  * ========================================================================
  */
 
+static int AndersonAccQRDelete(KINMem kin_mem, N_Vector* Q, sunrealtype* R,
+                               int depth)
+{
+  /* local dot product flag for single buffer reductions */
+  sunbooleantype dotprodSB = SUNFALSE;
+
+  if ((kin_mem->kin_vtemp2->ops->nvdotprodlocal ||
+       kin_mem->kin_vtemp2->ops->nvdotprodmultilocal) &&
+      kin_mem->kin_vtemp2->ops->nvdotprodmultiallreduce)
+  {
+    dotprodSB = SUNTRUE;
+  }
+
+  /* Delete left-most column vector from QR factorization */
+  sunrealtype a, b, temp, c, s;
+
+  for (int i = 0; i < depth - 1; i++)
+  {
+    a                          = R[(i + 1) * depth + i];
+    b                          = R[(i + 1) * depth + i + 1];
+    temp                       = SUNRsqrt(a * a + b * b);
+    c                          = a / temp;
+    s                          = b / temp;
+    R[(i + 1) * depth + i]     = temp;
+    R[(i + 1) * depth + i + 1] = ZERO;
+    /* OK to reuse temp */
+    if (i < depth - 1)
+    {
+      for (int j = i + 2; j < depth; j++)
+      {
+        a                    = R[j * depth + i];
+        b                    = R[j * depth + i + 1];
+        temp                 = c * a + s * b;
+        R[j * depth + i + 1] = -s * a + c * b;
+        R[j * depth + i]     = temp;
+      }
+    }
+    N_VLinearSum(c, Q[i], s, Q[i + 1], kin_mem->kin_vtemp2);
+    N_VLinearSum(-s, Q[i], c, Q[i + 1], Q[i + 1]);
+    N_VScale(ONE, kin_mem->kin_vtemp2, Q[i]);
+  }
+
+  /* Shift R to the left by one. */
+  for (int i = 1; i < depth; i++)
+  {
+    for (int j = 0; j < depth - 1; j++)
+    {
+      R[(i - 1) * depth + j] = R[i * depth + j];
+    }
```

Comment:
Optional suggestion: use `memmove`

Reply:
I'll add this to the issue for future enhancements